samples/bluetooth/iso_receive/src/main.c

```diff
@@ -0,0 +1,443 @@
+}
+
+static void iso_disconnected(struct bt_iso_chan *chan, uint8_t reason)
+{
+	printk("ISO Channel %p disconnected with reason 0x%02x\n",
+	       chan, reason);
+
+	if (reason != BT_HCI_ERR_OP_CANCELLED_BY_HOST) {
+		k_sem_give(&sem_big_sync_lost);
+	}
+}
+
+static struct bt_iso_chan_ops iso_ops = {
+	.recv		= iso_recv,
+	.connected	= iso_connected,
+	.disconnected	= iso_disconnected,
+};
+
+static struct bt_iso_chan_io_qos iso_rx_qos;
+
+static struct bt_iso_chan_qos bis_iso_qos = {
+	.rx = &iso_rx_qos,
+};
+
+static struct bt_iso_chan bis_iso_chan = {
+	.ops = &iso_ops,
+	.qos = &bis_iso_qos,
+};
+
+static struct bt_iso_chan *bis[BIS_ISO_CHAN_COUNT] = { &bis_iso_chan };
+
+static struct bt_iso_big_sync_param big_sync_param = {
+	.bis_channels = bis,
+	.num_bis = BIS_ISO_CHAN_COUNT,
+	.bis_bitfield = (BIT(BIS_ISO_CHAN_COUNT) - 1),
+	.mse = 1,
+	.sync_timeout = 100, /* in 10 ms units */
+};
+
+void main(void)
+{
+	struct bt_le_per_adv_sync_param sync_create_param;
+	struct bt_le_per_adv_sync *sync;
+	struct bt_iso_big *big;
+	uint32_t sem_timeout;
+	int err;
+
+	printk("Starting Synchronized Receiver Demo\n");
+
+#if defined(HAS_LED)
+	printk("Get reference to LED device...");
+	dev = device_get_binding(LED0);
+	if (!dev) {
+		printk("Failed.\n");
+		return;
+	}
+	printk("done.\n");
+
+	printk("Configure GPIO pin...");
+	err = gpio_pin_configure(dev, PIN, GPIO_OUTPUT_ACTIVE | FLAGS);
+	if (err) {
+		return;
+	}
+	printk("done.\n");
+
+	k_delayed_work_init(&blink_work, blink_timeout);
+#endif /* HAS_LED */
+
+	/* Initialize the Bluetooth Subsystem */
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return;
+	}
+
+	printk("Scan callbacks register...");
+	bt_le_scan_cb_register(&scan_callbacks);
+	printk("success.\n");
+
+	printk("Periodic Advertising callbacks register...");
+	bt_le_per_adv_sync_cb_register(&sync_callbacks);
+	printk("Success.\n");
+
+	do {
+		per_adv_lost = false;
+
+		printk("Start scanning...");
+		err = bt_le_scan_start(BT_LE_SCAN_CUSTOM, NULL);
+		if (err) {
+			printk("failed (err %d)\n", err);
+			return;
+		}
+		printk("success.\n");
+
+#if defined(HAS_LED)
+		printk("Start blinking LED...\n");
+		led_is_on = false;
+		gpio_pin_set(dev, PIN, (int)led_is_on);
+		k_delayed_work_submit(&blink_work, BLINK_ONOFF);
+#endif /* HAS_LED */
+
+		printk("Waiting for periodic advertising...\n");
+		per_adv_found = false;
+		err = k_sem_take(&sem_per_adv, K_FOREVER);
+		if (err) {
+			printk("failed (err %d)\n", err);
+			return;
+		}
+		printk("Found periodic advertising.\n");
+
+		printk("Stop scanning...");
+		err = bt_le_scan_stop();
+		if (err) {
+			printk("failed (err %d)\n", err);
+			return;
+		}
+		printk("success.\n");
+
+		printk("Creating Periodic Advertising Sync...");
+		bt_addr_le_copy(&sync_create_param.addr, &per_addr);
+		sync_create_param.options = 0;
+		sync_create_param.sid = per_sid;
+		sync_create_param.skip = 0;
+		sync_create_param.timeout = per_interval_ms * 2;
```

Comment:
The timeout is in 10 ms units, right?

```suggestion
		sync_create_param.timeout = per_interval_ms * 6 / 10 ;
```

Reply:
Correct. Also added a #define for the multiplier if we want to use the same for both this timeout and the sem timeout.